### ISSUE TITLE
Updating models?

### ISSUE THREAD

**Issue by igr** [vexy/Fridge#50]
What happens when a model gets updated?

It seems that Fridge fails to unfreeze existing data into a new structure. I see two cases here:

1. _extended_ model (e.g., new field added): I expected this to work when a new field has a default value.
2. _modified_ model (e.g., existing fields changes type etc): this is ok to fail.

One idea is to catch errors on load and then try to _import_ data by first loading them into a generic dictionary/map (whatever is available on swift) and then manually importing data into the new, extended structure. This process could also give a mapping callback that could be used in case of modified fields (e.g. `(fieldName, oldValue) -> newValue`)

**Comment by vexy** [comment]
> What happens when a model gets updated?

Unfortunately, Fridge fails to unfreeze a model that gets changed meanwhile. Reason for this unwanted behavior is internals of the _freezing_ sub-system (file storage).

As this may lead to extra responsibility of the storage layer and increase complexity - which is tried to be avoided by "moving" these tasks to a upper layers - this is known limitation.

However, this may be sort of _workaround_ solution:
- `drop` original model
- setup new (changed structure)
- `freeze` it again

Eventually, this may be wrapped into standalone function, named eg. `swap(..)` ?

**Comment by igr** [comment]
Yeah, agree - I was thinking of having it as an external tool, probably (mini-freezer:)
